devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -832,11 +832,16 @@ public void stop() throws InterruptedException {
         for (LocalProject project : localProject.getSelfWithLocalDeps()) {
             inProject.add(project.getKey());
             if (project.getClassesDir() != null &&
-            //if this project also contains Quarkus extensions we do no want to include these in the discovery
-            //a bit of an edge case, but if you try and include a sample project with your extension you will
-            //run into problems without this
-                    (Files.exists(project.getClassesDir().resolve("META-INF/quarkus-extension.properties")) ||
-                            Files.exists(project.getClassesDir().resolve("META-INF/quarkus-build-steps.list")))) {
+                    (
+                    // if it is not found among project.getArtifacts() it shouldn't be included (e.g. a local test dependency)
+                    localProject != project
```

Comment:
Are classifiers relevant here?

Reply:
It's interesting, I thought they should be checked too. But I couldn't get test scope deps into it. I guess it's related to the phase the dev mode is running in.
Javadoc says the content of `project.getArtifacts()` will depend on the phase and which dependencies have been resolved, iiuc. I tried `mvn clean test quarkus:dev`. That would actually run tests before the devmode. But the test dependencies still wouldn't be found in `project.getArtifactMap()`

Reply:
Actually, it's a good opportunity to check the type of the dependency. Sometimes people add dependencies on POMs, just for the transitive dependencies. I'll add that here and the check for the scope anyway.